src/components/ec-table-filter/ec-table-filter.vue

```diff
@@ -67,12 +72,16 @@ const props = defineProps({
     type: Boolean,
     default: false,
   },
+  hasStretchFilter: Boolean,
 });
 
 const emit = defineEmits(['update:modelValue', 'change']);
 
 const hasFilters = computed(() => !!Object.keys(props.modelValue).length);
 
+const hasFullWidthFilter = computed(() => (props.filters ? Object.values(props.filters).some(filter => filter.isFullWidth) : false));
+const stretchLayoutColumns = computed(() => (props.filters ? `repeat(${Object.values(props.filters).length - 1}, auto) 1fr auto` : ''));
```

Comment:
the problem I can see here is purge css process, tailwind used it before, now it doesn't, however it has a step which cleans up the bundle and remove all unused classes, we need to check whether it could affect this case.

https://tailwindcss.com/docs/content-configuration#dynamic-class-names

Reply:
How `v-bind` in CSS works is by creating an empty CSS var in the component style and update its value when needed. 
For example, lets say that `stretchLayoutColumns` creates a CSS var called `--XXXXX`, so:
1. add a style attribute with the empty var -> `<div class="..." style="--XXXXX: ;">...</div>`
2. when we use `v-bind` with `stretchLayoutColumns` within the component it will replace it for the CSS var -> 
```
&--has-stretch-filter {
   grid-template-columns: v-bind(stretchLayoutColumns);
}
/* ^^ This is transformed into:*/
&--has-stretch-filter {
   grid-template-columns: var(--XXXXX);
}
```
3. When `stretchLayoutColumns` value changes Vue updates the CSS var value from the style attribute

Even if the CSS var value is empty is being used, so purge CSS can't remove it. I just build Chameleon on my local to test it and here is how it looks in browser devtools:
![image](https://github.com/Ebury/chameleon/assets/77344353/fd29964d-d106-4338-af96-69ce868b6ae9)

Reply:
I see, yes it works, as it is not tailwind class. Sure. Sorry, for some reason I was in the context of tailwind classes considering this :).